app/modules/import-export/assets/js/admin.js

```diff
@@ -1,19 +1,245 @@
+			return;
+		}
+		this.addRevertBtnListener( revertButton );
+		this.maybeAddRevertBtnMargin( revertButton );
+	}
+
+	/**
+	 * AddRevertBtnListener
+	 *
+	 * @param {HTMLAnchorElement} revertButton
+	 */
+	addRevertBtnListener( revertButton ) {
+		revertButton.addEventListener( 'click', this.revertBtnOnClick.bind( this ) );
+	}
+
+	/**
+	 * MaybeAddRevertBtnMargin
+	 *
+	 * @param {HTMLAnchorElement} revertButton
+	 */
+	maybeAddRevertBtnMargin( revertButton ) {
+		const referrerKit = new URLSearchParams( revertButton.href ).get( 'referrer_kit' );
+		if ( ! referrerKit ) {
+			return;
+		}
+		revertButton.style.marginBottom = this.calculateMargin( revertButton );
+		this.scrollToBottom();
+	}
+
+	/**
+	 * CalculateMargin
+	 *
+	 * @param {HTMLAnchorElement} revertButton
+	 *
+	 * @return {string}
+	 */
+	calculateMargin( revertButton ) {
+		const adminBar = document.getElementById( 'wpadminbar' );
+		const adminBarHeight = adminBar ? adminBar.offsetHeight : 0;
+
+		const revertKitHeight = revertButton.parentElement.offsetHeight;
+
+		return (
+			document.body.clientHeight -
+			adminBarHeight -
+			revertKitHeight -
+			document.getElementById( 'wpfooter' ).offsetHeight -
+			15 // Extra margin at the top
+		) + 'px';
+	}
+
+	/**
+	 * Scroll to the bottom of the page
+	 */
+	scrollToBottom() {
+		setTimeout( () => {
+			window.scrollTo( 0, document.body.scrollHeight );
+		} );
+	}
+
+	/**
+	 * RevertBtnOnClick
+	 *
+	 * @param {Event} event
+	 */
+	revertBtnOnClick( event ) {
 		event.preventDefault();
 
+		const kitNameToRemove = this.getKitToRemoveName();
+
 		elementorCommon.dialogsManager.createWidget( 'confirm', {
-			headerMessage: __( 'Sure you want to make these changes?', 'elementor' ),
-			message: __( 'Removing assets or changing your site settings can drastically change the look of your website.', 'elementor' ),
+			headerMessage: __( 'Are you sure?', 'elementor' ),
+			message: __( 'Removing ', 'elementor' ) + kitNameToRemove + __( ' will permanently delete changes made to the Kit\'s content and site settings', 'elementor' ),
 			strings: {
-				confirm: __( 'Yes', 'elementor' ),
-				cancel: __( 'No, go back', 'elementor' ),
+				confirm: __( 'Delete', 'elementor' ),
+				cancel: __( 'Cancel', 'elementor' ),
 			},
-			onConfirm() {
-				location.href = revertButton.href;
+			onConfirm: this.onRevertConfirm.bind( this, event.target, kitNameToRemove ),
+		} ).show();
+	}
+
+	onRevertConfirm( revertBtn, kitNameToRemove ) {
+		const referrerKit = new URLSearchParams( revertBtn.href ).get( 'referrer_kit' );
+
+		this.setSessionStorage( Boolean( referrerKit ), referrerKit, kitNameToRemove );
+
+		location.href = revertBtn.href;
+	}
+
+	setSessionStorage( referred, referrerKit, kitNameToRemove ) {
+		sessionStorage.setItem( this.isKitReferredKey, referred ? 'yes' : 'no' );
+		if ( referred ) {
+			sessionStorage.setItem( this.kitReferrerKey, referrerKit );
+		}
+		sessionStorage.setItem( this.kitNameToRemoveKey, kitNameToRemove );
+	}
```

Comment:
About the third question, I used the `referred` because there are 3 possibilities when showing the referrer kit dialog:
1. Don't show the dialog (User is on the page but has not removed a kit)
2. show not referred dialog
3. show referred dialog

Perhaps I could use `referrerKit` in the same way and set it to an empty string when not referred but a kit has been reverted, but this seems more semantic